API/app/Http/Controllers/GoogleAuthenticationController.php

```diff
@@ -61,37 +63,31 @@ public function handle(Request $request, Closure $next): Response
                 $request->merge(['user_id' => $googleId]); // Add user_id to request
 
                 Log::info("User validated, setting cookie in request and returning...");
-    
-                // Allow original request to proceed
-                return $next($request);
-            } else {
-                return response()->json([
-                    'status' => 'error',
-                    'message' => 'Unauthorized: Invalid token or error occurred (contact administrator)',
-                ], 401);
-            }
-        } else {
-            // We are validating an existing cookie (check cookie instead)
-            $credential = $request->cookie('speedcart_auth');
-            Log::debug("Validating user sign in cookie with value " . $credential);
 
-            // Validate JWT stored in cookie
-            $googleId = $this->validateGoogleJwt($credential);
-            if ($googleId) {
-                $request->merge(['user_id' => $googleId]); // Add user_id to request
+                // This should work (consult official documentation for more)
+                $user = User::where('user_id', $googleId)->first();
+                if ($user) {
+                    Log::info("Logging in user");
```

Comment:
Also change this logging to only show on DEBUG mode (add constant to this file if necessary) and add $googleId variable to logging statement for better debugging.